scipy/stats/_multivariate.py

```diff
@@ -3850,3 +3851,405 @@ def rvs(self, dim, size=1, random_state=None):
+the shape matrix is the identity times that value, a vector of
+diagonal entries for the shape matrix, or a two-dimensional array_like.
+"""
+
+_mvt_doc_frozen_callparams_note = \
+"""See class definition for a detailed description of parameters."""
+
+mvt_docdict_params = {
+    '_mvt_doc_default_callparams': _mvt_doc_default_callparams,
+    '_mvt_doc_callparams_note': _mvt_doc_callparams_note,
+    '_doc_random_state': _doc_random_state
+}
+
+mvt_docdict_noparams = {
+    '_mvt_doc_default_callparams': "",
+    '_mvt_doc_callparams_note': _mvt_doc_frozen_callparams_note,
+    '_doc_random_state': _doc_random_state
+}
+
+
+class multivariate_t_gen(multi_rv_generic):
+    r"""
+    A multivariate t-distributed random variable.
+    The `loc` parameter specifies the location. The `shape` parameter specifies
+    the positive definite shape matrix. The `df` parameter specifies the
+    degrees of freedom.
+
+    In addition to calling the methods below, the object itself may be called
+    as a function to fix the location, shape matrix, and degrees of freedom
+    parameters, returning a "frozen" multivariate t-distribution random.
+
+    Methods
+    -------
+    ``pdf(x, loc=None, shape=1, df=1, allow_singular=False)``
+        Probability density function.
+    ``logpdf(x, loc=None, shape=1, df=1, allow_singular=False)``
+        Log of the probability density function.
+    ``rvs(loc=None, shape=1, df=1, size=1, random_state=None)``
+        Draw random samples from a multivariate t-distribution.
+
+    Parameters
+    ----------
+    x : array_like
+        Quantiles, with the last axis of `x` denoting the components.
+    %(_mvt_doc_default_callparams)s
+    %(_doc_random_state)s
+
+    Notes
+    -----
+    %(_mvt_doc_callparams_note)s
+    The matrix `shape` must be a (symmetric) positive definite matrix. The
+    determinant and inverse of `shape` are computed as the pseudo-determinant
+    and pseudo-inverse, respectively, so that `shape` does not need to have
+    full rank.
+
+    The probability density function for `multivariate_t` is
+
+    .. math::
+
+        f(x) = \frac{\Gamma(\nu + p)/2}{\Gamma(\nu/2)\nu^{p/2}\pi^{p/2}|\Sigma|^{1/2}}
+               \exp\left[1 + \frac{1}{\nu} (\mathbf{x} - \boldsymbol{\mu})^{\top}
+               \boldsymbol{\Sigma}^{-1}
+               (\mathbf{x} - \boldsymbol{\mu}) \right]^{-(\nu + p)/2},
+
+    where :math:`p` is the dimension of :math:`\mathbf{x}`,
+    :math:`\boldsymbol{\mu}` is the :math:`p`-dimensional location,
+    :math:`\boldsymbol{\Sigma}` the :math:`p \times p`-dimensional shape
+    matrix, and :math:`\nu` is the degrees of freedom.
+
+    .. versionadded:: 1.5.0
+
+    Examples
+    --------
+    >>> import matplotlib.pyplot as plt
+    >>> from scipy.stats import multivariate_t
+    >>> x, y = np.mgrid[-1:1:.01, -1:1:.01]
+    >>> pos = np.dstack((x, y))
+    >>> rv = multivariate_t([1.0, -0.5], [[2.1, 0.3], [0.3, 1.5]], df=2)
+    >>> plt.contourf(x, y, rv.pdf(pos))
```

Comment:
This renders as:
![image](https://user-images.githubusercontent.com/6570539/93381445-88b84200-f815-11ea-9429-7010b02c5b00.png)

Might want to make the aspect ratio square and choose a more complete window.

Reply:
Okay, I made the window more complete.

To my knowledge, changing the aspect ratio would require making an axes and then setting it, which would add a lot of boilerplate to the small example IMO. I just followed the contour plot [here](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.multivariate_normal.html#scipy.stats.multivariate_normal).

Reply:
@mdhaber, thoughts? Here is the current version with a more complete window:

![](https://user-images.githubusercontent.com/2818208/94142371-2cb87380-fe3c-11ea-9dd7-06eda57004a4.png)

To make the aspect ratio square, we would need to add two lines to the example:

```python
>>> import numpy as np
>>> import matplotlib.pyplot as plt
>>> from scipy.stats import multivariate_t
>>> x, y = np.mgrid[-1:3:.01, -2:1.5:.01]
>>> pos = np.dstack((x, y))
>>> rv = multivariate_t([1.0, -0.5], [[2.1, 0.3], [0.3, 1.5]], df=2)
>>> fig, ax = plt.subplots(1, 1)                                            # This line
>>> ax.set_aspect('equal')                                                  # And this line
>>> plt.contourf(x, y, rv.pdf(pos))
```

Reply:
I thought it would be nice to keep it square so that the effect of the shape parameter is visually clear.
I am OK with adding those two lines to do that. 
@rlucas7 is it worth it?

Reply:
> I thought it would be nice to keep it square so that the effect of the shape parameter is visually clear.
> I am OK with adding those two lines to do that.
> @rlucas7 is it worth it?

I do think that visually getting a full equi-contour trace is helpful. 
Especially for users who are less familiar w/the distribution. 
The rendering will be helpful to refer to if we get questions and upshot is the figure stays within line char limit. 
We can omit the two end of line comments:

```
>>> fig, ax = plt.subplots(1, 1)                                            # This line
>>> ax.set_aspect('equal')                                                  # And this line
```
and keep the rest of the two added lines.
Unless anyone has major objections, let's use the manual axis.

Reply:
Okay, I've added those two lines.

Reply:
ok, I'm marking as resolved @mdhaber if you had other concerns on this part please reopen.